lib/src/google_fonts_descriptor.dart

```diff
@@ -9,10 +13,23 @@ import 'google_fonts_family_with_variant.dart';
 class GoogleFontsDescriptor {
   const GoogleFontsDescriptor({
     @required this.familyWithVariant,
-    @required this.fontUrl,
+    @required this.file,
   })  : assert(familyWithVariant != null),
-        assert(fontUrl != null);
+        assert(file != null);
 
   final GoogleFontsFamilyWithVariant familyWithVariant;
-  final String fontUrl;
+  final GoogleFontsFile file;
+}
+
+/// Describes a font file as it is _expected_ to be received form the server.
+///
+/// If a file is retrieved and its hash does not match [expectedFileHash], or it
+/// is not of [expectedLength] bytes length, the file will be rejected.
+class GoogleFontsFile {
+  GoogleFontsFile(this.expectedFileHash, this.expectedLength);
+
+  final String expectedFileHash;
+  final int expectedLength;
+
+  String get url => 'https://fonts.gstatic.com/s/a/$expectedFileHash.ttf';
```

Comment:
This is string is also interpolated in the generator. It should probably be in a reusable function so that the url part is only typed in a single location

Reply:
Good point, I updated the generator to use this getter here.

Reply:
After trying this a couple of times, there's actually no way to share this logic I believe. 

This is because _dart_ scripts can not import anything from flutter. And since `google_fonts_descriptor.dart` imports flutter code, our dart generator can not import it. The only way I can think to solve this would be to separate out logic into a separate local package that both the generator and the library depended on. For now though, I'm just going to leave the URL logic in both places.